client/core/core.go

```diff
@@ -2472,11 +2476,77 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	}
 	tracker.matchMtx.RUnlock()
 	if revokedMatch == nil {
-		return fmt.Errorf("no match found with id %s for order %s", matchID, oid.String())
+		return fmt.Errorf("no match found with id %s for order %v", matchID, oid)
 	}
 
+	errs := newErrorSet("handleRevokeMatchMsg (order %v, match %v): ", oid, matchID)
+
+	// Set the match as revoked.
 	revokedMatch.MetaMatch.MetaData.Proof.IsRevoked = true
-	return tracker.db.UpdateMatch(&revokedMatch.MetaMatch)
+	err = tracker.db.UpdateMatch(&revokedMatch.MetaMatch)
+	if err != nil {
+		errs.add("unable to update match: %v", err)
+	}
+
+	// Notify the user of the failed match.
+	corder, _ := tracker.coreOrder() // no cancel order
+	tracker.notify(newOrderNote("Match revoked", fmt.Sprintf("Match %s has been revoked",
+		tracker.token()), db.WarningLevel, corder))
+
+	// Also set the order as revoked.
+	metaOrder := tracker.metaOrder()
+	metaOrder.MetaData.Status = order.OrderStatusRevoked
+	err = tracker.db.UpdateOrder(metaOrder)
+	if err != nil {
+		errs.add("unable to update order: %v", err)
+	}
+
+	// Notify the user of the revoked order.
+	details := fmt.Sprintf("%s order on %s-%s at %s has been revoked (%s)",
+		strings.Title(sellString(tracker.Trade().Sell)), unbip(tracker.Base()),
+		unbip(tracker.Quote()), dc.acct.host, tracker.token())
+	tracker.notify(newOrderNote("Order revoked", details, db.WarningLevel, corder))
+
+	// Send out a data notification with the revoke information.
+	cancelOrder := &Order{
+		Host:     tracker.dc.acct.host,
+		MarketID: tracker.mktID,
+		Type:     order.CancelOrderType,
+		Stamp:    encode.UnixMilliU(time.Now()),
+		Epoch:    corder.Epoch,
+		TargetID: corder.ID, // the important part for the frontend
+	}
+	tracker.notify(newOrderNote("revoke", "", db.Data, cancelOrder))
+
+	// Unlock funding coins if they are not already unlocked. Do this one at a
+	// time if wallet fails if just one cannot be unlocked.
+	for _, coin := range tracker.coins {
+		err = tracker.wallets.fromWallet.ReturnCoins(asset.Coins{coin})
+		if err != nil {
+			log.Warnf("unable to unlock coin %v: %v", coin, err)
+		}
+	}
+
+	// Tick asset to trigger recovery if the match was revoked during
+	// swap and either refund or redeem is needed. Or not?
+	counts, err := tracker.tick()
+	if err != nil {
+		errs.addErr(err)
+	}
+	if len(counts) > 0 {
+		dc.refreshMarkets()
+		c.updateBalances(counts)
+	}
```

Comment:
Need input on this.  The handler updates the match and order statuses, sends notes to the frontend, and acks to the DEX.  However, it's not clear to me what refresh/update operations also need to be run.

Reply:
Yep. I think `updateAssetBalance` and probably a `refreshUser`. This might be a good time to mention that #379 is still open and I'm looking for a better solution for user state updates.